posthog/hogql_queries/events_query_runner.py

```diff
@@ -263,10 +263,10 @@ def apply_dashboard_filters(self, dashboard_filter: DashboardFilter):
         if dashboard_filter.properties:
             self.query.properties = (self.query.properties or []) + dashboard_filter.properties
 
-    def columns(self, query_columns: list | None) -> list[str]:
-        columns = query_columns or []
+    def columns(self, result_columns: list | None) -> list[str]:
+        columns = result_columns or []
         return [
-            col if col == "*" or "person" else columns[idx] if len(columns) > idx else col
+            columns[idx] if len(columns) > idx and " as " in col else col
```

Comment:
@mariusandra I wasn't super happy doing this but anything else I tried seemed to have a bunch of knock on effects and failing tests e.g. `query_result.columns` would return `$lib` where the frontend expects `properties.$lib` to correctly render it as a property column. Mostly caused by the default columns which seem to be used in a bunch of places:
https://github.com/PostHog/posthog/blob/5754981a3d500c19a49da72948512ed3182385e2/frontend/src/queries/nodes/DataTable/utils.ts#L4-L11

I opted for the simpler route of just affecting columns with an alias but there is certainly improvements that could be made here from an implementor with a little more context on Product Analytics. Not urgent, I'm certainly unblocked 😄